---
title: C26111
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C26111
helpviewer_keywords:
- C26111
ms.assetid: 85fc740a-3bbb-41b8-a848-95e243a08da9
author: mikeblome
ms.author: mblome
manager: wpickett
ms.workload:
- multiple
ms.openlocfilehash: 5541ec5dcc8ccd55fe5288265090e3d9c5958e04
ms.sourcegitcommit: 94b3a052fb1229c7e7f8804b09c1d403385c7630
ms.translationtype: MT
ms.contentlocale: ko-KR
ms.lasthandoff: 04/23/2019
ms.locfileid: "62550177"
---
# <a name="c26111"></a>C26111
warning C26111: 호출자가 잠금을 해제 하지 못함 \<잠금 > 함수를 호출 하기 전에 \<func >.

 `_Requires_lock_not_held_` 주석에는 함수가 호출될 때 지정된 잠금에 대한 잠금 횟수가 0보다 클 수 없다는 사전 조건이 적용됩니다. 다른 함수를 호출하기 전에 함수가 잠금을 해제하지 못할 경우 경고 C26111이 발생합니다.

## <a name="example"></a>예제
 다음 예제에서는 잠긴 섹션 내에서 `_Requires_lock_not_held_`에 대한 호출로 인해 `DoNotLock` 사전 조건이 위반되기 때문에 경고 C26111이 발생합니다.

```

typedef struct _DATA
{
    CRITICAL_SECTION cs;
    int d;
} DATA;

_Requires_lock_not_held_(p->cs)

void DoNotLock(DATA* p)
{
    EnterCriticalSection(&p->cs);
    p->d = 0;
    LeaveCriticalSection(&p->cs);
}

void LockedFunction(DATA* p)
{
    EnterCriticalSection(&p->cs);
    DoNotLock(p); // Warning C26111
    LeaveCriticalSection(&p->cs);
}
```